### PR TITLE
fv3 changes for s2s one step cold start 

### PR DESCRIPTION
**Pull request by junwang-noaa**
## Description

This PR contains fv3 code changes to set up on step cold start.
1) fv3 is now set up the bottom layer coupled fields at the end of initialization step. 
2) previous diagnostic subroutine Dump_cplFields is renamed to diagnose_cplFields.
3) Debug option -link_mpi=dbg is added to debug tests.

### Issue(s) addressed

- see  [issue 168](https://github.com/ufs-community/ufs-s2s-model/issues/168)

## Testing

RT will be conducted on wcoss, hera and orion.

## Dependencies

- waiting on [fv3atm PR#168](https://github.com/NOAA-EMC/fv3atm/pull/168)
- related [s2s PR #176](https://github.com/ufs-community/ufs-s2s-model/pull/176)

### CODE REVIEW

File: cmake/Intel.cmake [hunk 1]
```diff
@@ -40,6 +40,15 @@ if(REPRO)
...
 elseif(DEBUG)
     set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -traceback -ftrapuv")
+    if(DEBUG_LINKMPI)
+      if(OPENMP)
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -link_mpi=dbg_mt")
+      else()
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -link_mpi=dbg")
+      endif()
+    else()
+      set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
```
Comment (DusanJovic-NOAA) [inline cmake/Intel.cmake]: This line redundant. Remove lines 49 and 50

**Comment by climbfuji** [comment]
@junwang-noaa should I run tests on cheyenne. intel with both gnumake and cmake?

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by junwang-noaa** [comment]
I did not run gnumake in ufs-weather model, only cmake. I did not add
link_mpi option to compile configuration files (conf/*) on platforms other
than wcoss, hera and orion, you may want to check if it is working on
cheyenne.

**Comment by climbfuji** [comment]
> I did not run gnumake in ufs-weather model, only cmake. I did not add link_mpi option to compile configuration files (conf/*) on platforms other than wcoss, hera and orion, you may want to check if it is working on cheyenne.
> […](#)

In fact, you added it to `conf/configure.fv3.cheyenne.intel`, that's why I was asking. I'll test it!

**Comment by junwang-noaa** [comment]
Great, I did not test it.

**Comment by climbfuji** [comment]
> Great, I did not test it.
> […](#)

I think we need to tiny changes to `conf/configure.fv3.cheyenne.intel` and `conf/configure.fv3.cheyenne.gnu`, not directly related to this PR. There are missing dependencies on `POST_LIB` with gnumake that my usual regression testing using ecflow/cmake didn't catch.

The rest is looking good, although the regression tests are still running.

**Comment by climbfuji** [comment]
> > Great, I did not test it.
> > […](#)
> 
> I think we need to tiny changes to `conf/configure.fv3.cheyenne.intel` and `conf/configure.fv3.cheyenne.gnu`, not directly related to this PR. There are missing dependencies on `POST_LIB` with gnumake that my usual regression testing using ecflow/cmake didn't catch.
> 
> The rest is looking good, although the regression tests are still running.

Update: all debug compile jobs fail on cheyenne, because even though cheyenne.intel is using the Intel compiler, it is not using Intel MPI, but SGI MPT. We need to add the same logic to the cmake and gnumake config that you added for wcoss. And probably need the same for gaea, too. I can make those changes and send you a PR to update your branch. This will not require rerunning the regression tests, since these are only machine-dependent configuration changes for machines other than tier-1 platforms.

**Comment by junwang-noaa** [comment]
Sure.

**Review by climbfuji — COMMENTED** [review]
This looks good, just need to update the fv3atm submodule pointer and revert the `.gitmodules` changes.

**Review by climbfuji — APPROVED** [review]

**Pull request merged**